File: plog/src/main/java/com/blackbox/plog/utils/DateControl.java
package com.blackbox.plog.utils;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;


/**
 * Created by Umair Adil on 06/04/2017.
 */

public class DateControl {

    private String TAG = DateControl.class.getSimpleName();

    private static final DateControl ourInstance = new DateControl();

    public static DateControl getInstance() {
        return ourInstance;
    }

    public String getToday() {
        Date currentTime = new Date(System.currentTimeMillis());
        SimpleDateFormat sdf = new SimpleDateFormat("ddMMyyyy", Locale.ENGLISH);
        return sdf.format(currentTime);
    }

    public String getLastWeek() {
        Calendar cal = Calendar.getInstance();
        cal.setTime(new Date());
        cal.add(Calendar.DAY_OF_YEAR, -7);
        Date date = cal.getTime();
        SimpleDateFormat sdf = new SimpleDateFormat("dd", Locale.ENGLISH);
        return sdf.format(date);
    }

    public String getLastDay() {
        Calendar cal = Calendar.getInstance();
        cal.setTime(new Date());
        cal.add(Calendar.DAY_OF_YEAR, -1);
        Date date = cal.getTime();
        SimpleDateFormat sdf = new SimpleDateFormat("dd", Locale.ENGLISH);
        return sdf.format(date);
    }

    public String getHour() {
        Date currentTime = new Date(System.currentTimeMillis());
        SimpleDateFormat sdf = new SimpleDateFormat("HH", Locale.ENGLISH);
        return sdf.format(currentTime);
    }

    public String getCurrentDate() {
        Date currentTime = new Date(System.currentTimeMillis());
        SimpleDateFormat sdf = new SimpleDateFormat("dd", Locale.ENGLISH);
        return sdf.format(currentTime);
    }
}

File: plog/src/main/java/com/blackbox/plog/utils/DateControl.kt
package com.blackbox.plog.utils

import androidx.annotation.Keep
import com.blackbox.plog.pLogs.formatter.TimeStampFormat
import java.text.SimpleDateFormat
import java.util.*


/**
 * Created by Umair Adil on 06/04/2017.
 */
@Keep
class DateControl {

    private val TAG = DateControl::class.java.simpleName

    val today: String
        get() {
            val currentTime = Date(System.currentTimeMillis())
            val sdf = SimpleDateFormat(TimeStampFormat.DATE_FORMAT_1, Locale.ENGLISH)
            return sdf.format(currentTime)
        }

    val lastWeek: String
        get() {
            val cal = Calendar.getInstance()
            cal.time = Date()
            cal.add(Calendar.DAY_OF_YEAR, -7)
            val date = cal.time
            val sdf = SimpleDateFormat("dd", Locale.ENGLISH)
            return sdf.format(date)
        }

    val lastDay: String
        get() {
            val cal = Calendar.getInstance()
            cal.time = Date()
            cal.add(Calendar.DAY_OF_YEAR, -1)
            val date = cal.time
            val sdf = SimpleDateFormat("dd", Locale.ENGLISH)
            return sdf.format(date)
        }

    val hour: String
        get() {
            val currentTime = Date(System.currentTimeMillis())
            val sdf = SimpleDateFormat("HH", Locale.ENGLISH)
            return sdf.format(currentTime)
        }

    val time: String
        get() {
            val currentTime = Date(System.currentTimeMillis())
            val sdf = SimpleDateFormat("HH_mm", Locale.ENGLISH)
            return sdf.format(currentTime)
        }

    val currentDate: String
        get() {
            val currentTime = Date(System.currentTimeMillis())
            val sdf = SimpleDateFormat("dd", Locale.ENGLISH)
            return sdf.format(currentTime)
        }

    companion object {

        val instance = DateControl()
    }
}
